### ISSUE TITLE
Build fails with non-existing out-of-tree REPOSITORY_LOC due to swig+cpp's POST_BUILD

### ISSUE THREAD

**Issue by jktjkt** [sysrepo/sysrepo#828]
Here's how a from-the-scratch shared-nothing builds fails:
```
2017-04-28 16:15:19,282 [output] FAILED: swig/libSysrepo-cpp.so 
2017-04-28 16:15:19,282 [output] : 
&& /opt/rh/devtoolset-6/root/usr/bin/c++  -fPIC -Wall -Wpedantic -std=c++11 -fno-strict-aliasing -g -O0   -shared -Wl,-soname,libSysrepo-cpp.so -o swig/libSysrepo-cpp.so swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Sysrepo.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Connection.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Session.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Struct.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Xpath.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Tree.cpp.o swig/CMakeFiles/Sysrepo-cpp.dir/cpp/src/Internal.cpp.o  src/libsysrepo.so -lrt -lpthread /home/turbo-hipster/target/lib/libredblack.so -lev -lprotobuf-c /home/turbo-hipster/target/lib64/libyang.so -Wl,-rpath,/home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/build/src:/home/turbo-hipster/target/lib:/home/turbo-hipster/target/lib64: 
&& cd /home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/build/swig 
&& /home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/build/src/sysrepoctl --install --yang=/home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/swig/cpp/tests/yang/swig-test-cpp-changes@2017-03-09.yang 
&& cd /home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/build/swig
&& /home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/build/src/sysrepoctl --install --yang=/home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/swig/cpp/tests/yang/swig-test-cpp-operations@2017-03-09.yang
2017-04-28 16:15:19,282 [output] [ERR] (srctl_ly_log_cb:487) libyang: Unable to use search directory "/home/turbo-hipster/target/etc-sysrepo/yang/" (No such file or directory)
2017-04-28 16:15:19,282 [output] [ERR] (md_init:1024) Unable to allocate memory in md_init
2017-04-28 16:15:19,283 [output] Error: Failed to initialize module dependencies context.
2017-04-28 16:15:19,283 [output] Installing a new module from file '/home/turbo-hipster/jobs/e8d4791723d1/30/230/1/check/check-czechlight-devtoolset6-el7/1cab956/sysrepo/sysrepo/swig/cpp/tests/yang/swig-test-cpp-changes@2017-03-09.yang'...
2017-04-28 16:15:19,283 [output] Install operation failed.
2017-04-28 16:15:19,283 [output] ninja: build stopped: subcommand failed.
```
This is because the build system calls ``sysrepoctl`` prior to ``make install`` via CMake's ``POST_BUILD`` feature.

I understand the motivation; perhaps it would be reasonable to create that directory at first?

**Comment by mislavn** [comment]
@jktjkt the pull request https://github.com/sysrepo/sysrepo/pull/831 should fix the issue, can you confirm it.

**Comment by mislavn** [comment]
Fixed with #831.
